packages/yarnpkg-pnp/sources/loader/makeApi.ts

```diff
@@ -172,9 +173,51 @@ export function makeApi(runtimeState: RuntimeState, opts: MakeApiOptions): PnpAp
   }
 
   /**
-   * Implements the node resolution for folder access and extension selection
+   * Implements the node resolution for the "exports" field
+   *
+   * @returns The remapped path or `null` if the package doesn't have an "exports" field
    */
+  function applyNodeExportsResolution(unqualifiedPath: PortablePath) {
+    const locator = findPackageLocator(ppath.join(unqualifiedPath, `internal.js` as Filename));
+    if (locator === null) {
+      throw makeError(
+        ErrorCode.API_ERROR,
+        `The resolveUnqualifiedExport function must be called with a valid unqualifiedPath`,
+      );
+    }
+
+    const {packageLocation} = getPackageInformationSafe(locator);
+
+    const pkgJson = JSON.parse(opts.fakeFs.readFileSync(ppath.join(packageLocation, `package.json` as Filename), `utf8`));
+
+    let subpath = ppath.contains(packageLocation, unqualifiedPath);
+    if (subpath === null) {
+      throw makeError(
+        ErrorCode.INTERNAL,
+        `unqualifiedPath doesn't contain the packageLocation (this is probably an internal error)`,
+      );
+    }
+
+    if (!/^\.{0,2}\//.test(subpath))
+      subpath = `./${subpath}` as PortablePath;
+
+    const resolvedExport = resolveExport(pkgJson, ppath.normalize(subpath), {
+      require: true,
+      browser: false,
+      // TODO: implement support for the --conditions flag
+      // Waiting on https://github.com/nodejs/node/issues/36935
+      conditions: [],
```

Comment:
Should be allowed to pass this through `resolveUnqualifiedExport` for API consumers like the ESM loader in #2161